website/blog/2023-11-15-run-webassembly-wasm-workloads-windows-and-macos.md

```diff
@@ -0,0 +1,342 @@
+
+The Wasm binary format was designed to be secure. Wasm modules are isolated from the rest of the system, and they cannot access any system resources without explicit permission. This makes Wasm modules very safe to run, even in untrusted environments. But on another hand, for developing backend applications, this restriction is limiting the usage of WASM.
+
+## The extension of WebAssembly
+
+WebAssembly System Interface (WASI) was born as an essential complement to WebAssembly.
+
+It is a system interface that extends WebAssembly's capabilities beyond the browser, making it suitable for a wider range of environments, including servers, edge devices, and more.
+
+While with WASM you had limited access to the host resources, WASI provides a standard set of system calls, enabling WebAssembly modules to interact with the host operating system in a secure and consistent manner: it includes filesystem access, sockets, and other low-level resources.
+
+## Running WebAssembly outside the browser
+
+Wasm has shipped in the major browser engines so the usage of wasm is possible without any 3rd party addition in the browser land. But when it comes to the edge/system usage, you need to find a virtual machine to run these workloads supporting WASI extension. And there is not only one application to run them, there are several wasm runtime such as WasmEdge, Wasmtime, Wasmer, and so on. All runtimes support different CPU architectures.
+
+Since WASI is still maturing some of the API provided in these runtimes has not reached the standard, so users need to be careful to write portable applications that do not depend on a given runtime.
+
+Besides running wasm/WASI workloads on your computer, there is also the question of how you package this binary format, share, and distribute it. A convenient way to distribute and run these workloads is to use OCI images as it provides all the basics: package, storage and distribution of the binaries. Then comes the execution part.
+
+## Using Podman engine with wasm
+
+When using containers with Podman on macOS or Windows, you have a virtual machine called a "Podman machine" that is executing a Linux environment. We need to add support for WASM inside this Linux environment. Podman is using the crun project as its OCI runtime, so crun needs to be able to run or delegate execution to WASM runtimes. Lucky for us, crun supports WASM execution.
+
+From the user's point of view, support for WASM is provided as an additional platform. So when executing a WASM workload, we specify as a platform `--platform=wasi/wasm` instead of for example `--platform=linux/arm64` or `--platform=linux/amd64`
```

Comment:
```suggestion
From the user's point of view, support for WASM is provided as an additional platform. So when executing a WASM workload, we specify as a platform `--platform=wasi/wasm` instead of for example `--platform=linux/arm64` or `--platform=linux/amd64`.
```